docs/reference/async-processing.mdx

```diff
@@ -0,0 +1,116 @@
+
+It works by giving each StreamThread a dedicated thread pool which it uses to process the records passing through an async processor. Records of the same key are still processed in order and all the same consistency and correctness guarantees of Kafka Streams are maintained, including exactly-once semantics.
+
+While async processing can help improve performance of any application with a large keyspace, it was built to shine when processors include a long-running step such as remote state store requests or generic RPCs. The classic solution for achieving high throughput when you are bottlenecked by network latency is to increase the parallelism of the calls to the target system, which is exactly what the async processor does for Kafka Streams. 
+
+This makes the Responsive Async Processor the perfect tool to complement Responsive state stores, meaning you don’t need to sacrifice performance in order to gain the availability and elasticity benefits of remote storage with Kafka Streams!
+
+## Expected Performance Gains
+
+As with all real-world systems, performance depends heavily on the specific use case and parameters. However, in all of our experiments, async processing showed an improvement over regular Kafka Streams when a topology includes a processor which makes a remote call. 
+
+<img
+    alt={"Responsive Asyc Processor Peformancs"}
+    src={require('/static/img/docs/reference/async-processing/async-processing-perf.png').default}
+    className={'px-[10%] mb-4'} />
+
+
+To give an idea of the benefits of using the async processor when your topology includes higher latency external calls, we ran a test that injected a variable latency in our remote storage client while running a simple application that would deduplicate events over a window. A couple of things stand out from the results above: 
+
+1. The greater the remote latency, the bigger the impact of async. At 2 ms round trip latency, throughput improves by 5.5x. At 1 ms round trip latency, throughput improves by 3x.
+2. The throughput with async processing enabled is stable even with varying latency.
+
+:::note
+The async processor makes the round trip network latency a non-factor in achieving high throughput. However, using the async processor will generally place more load on the remote system, which in turn may get saturated, which in turns will drive up the latencies, which will bring down the throughput gains of the async processor. 
+
+Even so, the async processor performed at a steady 30% better even when the remote system was at its limit in our testing.
+:::
+
+We are continuing to test and tune the async processor, and will share more results as we have them. 
+
+## Enabling async processing
+
+Turning async processing on (or off!) is quick and easy. All it requires is one line of code and one config and you’re good to go! Make sure to check out the “Limitations” section for what is covered by the current version and what will be coming in a future release.
```

Comment:
@ableegoldman can we wrap our async processor in a wrapper that would allow you to turn it totally off via config? basically if you set the num threads to 1 it doesn't use any of the async mechanisms and just defaults back to the basic behavior

Reply:
that's a good idea -- yeah it shouldn't be too hard (famous last words). I'll give it a shot now since we're waiting on one more fix before cutting the new release

Reply:
ok it's actually not simple at all haha 😭  -- the usual culprit of "configs aren't available at topology build time". It's still possible just not as easy as I was hoping -- I'll timebox this and file a ticket to follow up on if I can't get it working in ~30 minutes or so

Reply:
Without this it may actually be harder for customers to test/bake in a staging environment. I know every customer has per-env configs but most don't have per-env code deployments.

Reply:
Another option here would be to set the async threads to 1 and then make the async processor a no-op.